### ISSUE TITLE
Search result screen reader accessibility issues

### ISSUE THREAD

**Issue by robertknight** [hypothesis/h#4557]
The search results shown in activity pages at https://hypothes.is/search are difficult to navigate with a screen reader. Testing with VoiceOver on OS X 10.12 I see the following issues:

1. Bucket titles are not links or buttons, so there is no indication that they can be clicked. There is also no indication of the expanded/collapsed state for a bucket. See https://github.com/hypothesis/h/pull/4554 and in particular the Google Doc linked in that PR's description.
1. Annotation cards are not labeled as a group
1. The elements of the annotation card do not have generic labels, so it isn't clear that eg. the username is a username. Specifically: The username, date, group name, body and quote do not have generic labels.
1. The annotation count is not labeled, so the screen reader just reads the number instead of saying something like "42 annotations on this document".

CC @judell

Several of these issues are fixed by https://github.com/hypothesis/h/pull/4554. In this issue I'm just describing the problems that I encountered with light testing of that page.
